### DISCUSSION TITLE
how RMB would respond to a sender in case of a message expire in backlog

### DISCUSSION THREAD

**Discussion by sameh-farouk** [threefoldtech/rmb-rs#7] [General]
> if $dst != local, this message is intended for remote process The message is then stored in redis on key (backlog.$id) AND a TTL is set on that with $exp. This means a message that never receive a response will eventually be flushed out of redis memory. Then the message id ($id) is pushed to msgbus.system.forward.

this is a better approach indeed than the one we have and would eliminate the need for do checking the expiration of the messages and clear expired messages from REDIS on the RMB logic itself, especially that EXPIRE was improved since Redis 6.0 and a lot of previous limitations of EXPIRE was enhanced and dealt with.

with that said, I want to discuss how RMB would respond to the sender in case of a message expire, would it subscribe to EXPIRED events and handle the events stream, I think this is not possible as we won't be able to retrieve the message to know to whom we will respond (we would have the $Id from the key name only but the message would be gone?).

**Comment by muhamadazmy** [comment]
It won't clients need to timeout on their own like how all clients do. I already explained why making rmb send 'timeout' messages to client was a bad idea from start https://github.com/threefoldtech/rmb-rs/discussions/6#discussioncomment-2523269
